cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/jmx/JmxConnect.java

```diff
@@ -0,0 +1,251 @@
+ *
+ * @author Robert Stupp
+ */
+public class JmxConnect implements Closeable {
+    private static final String DEFATULT_CASSANDRA_JMX_HOST = "127.0.0.1";
+    private static final int DEFAULT_CASSANDRA_JMX_PORT = 7199;
+
+    private static final String STORAGE_SERVICE_NAME = "org.apache.cassandra.db:type=StorageService";
+    private static final String ENDPOINT_SNITCH_INFO_NAME = "org.apache.cassandra.db:type=EndpointSnitchInfo";
+
+    private static final String fmtUrl = "service:jmx:rmi:///jndi/rmi://[%s]:%d/jmxrmi";
+
+    private static final String GOSSIPER_MBEAN_NAME = "org.apache.cassandra.net:type=Gossiper";
+
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+
+    private JMXConnector jmxc;
+    private MBeanServerConnection mbeanServerConn;
+
+    private StorageServiceMBean ssProxy;
+    private MessagingServiceMBean msProxy;
+    private StreamManagerMBean streamProxy;
+    private CompactionManagerMBean compactionProxy;
+    private FailureDetectorMBean fdProxy;
+    private CacheServiceMBean cacheService;
+    private StorageProxyMBean spProxy;
+    private HintedHandOffManagerMBean hhProxy;
+    private GCInspectorMXBean gcProxy;
+    private GossiperMBean gossProxy;
+    private EndpointSnitchInfoMBean snitchProxy;
+    private MemoryMXBean memProxy;
+    private RuntimeMXBean runtimeProxy;
+
+    private final ReentrantLock lock = new ReentrantLock();
+
+    public JmxConnect(String host, int port, String username, String password) {
+        if (host == null || host.trim().isEmpty())
+            host = DEFATULT_CASSANDRA_JMX_HOST;
+        if (port <= 0)
+            port = DEFAULT_CASSANDRA_JMX_PORT;
+
+        this.host = host;
+        this.port = port;
+        this.username = username;
+        this.password = password;
+    }
+
+    public JmxConnect(String host, int port) {
+        this(host, port, null, null);
+    }
+
+    @Override
+    public void close() throws IOException {
+        lock.lock();
+        try {
+            if (jmxc == null)
+                return;
+
+            jmxc.close();
+        } finally {
+            lock.unlock();
+
+            jmxc = null;
+            mbeanServerConn = null;
+            ssProxy = null;
+            msProxy = null;
+            streamProxy = null;
+            compactionProxy = null;
+            fdProxy = null;
+            cacheService = null;
+            spProxy = null;
+            hhProxy = null;
+            gcProxy = null;
+            gossProxy = null;
+            snitchProxy = null;
+            memProxy = null;
+            runtimeProxy = null;
+        }
+    }
+
+    public void connect() throws IOException {
+        lock.lock();
+        try {
+            if (jmxc != null)
+                return;
+
+            JMXServiceURL jmxUrl = new JMXServiceURL(String.format(fmtUrl, host, port));
+            Map<String, Object> env = new HashMap<String, Object>();
```

Comment:
Java 1.7 Style: Unnecessary generic args to HashMap constructor.
`new HashMap<>();`